runtime/compiler/control/J9CompilationStrategy.cpp

```diff
@@ -529,11 +646,13 @@ J9::CompilationStrategy::ProcessJittedSample::initializeRecompRelatedFields()
 
    if (_logSampling)
       {
-      _curMsg += sprintf(_curMsg, " cnt=%d ncl=%d glblSmplCnt=%d startCnt=%d[-%u,+%u] samples=[%d %d] windows=[%d %u] crtSmplIntrvlCnt=%u",
+      size_t maxMsgLen = sizeof(_msg) - (_curMsg - _msg);
+      int msgLen = snprintf(_curMsg, maxMsgLen, " cnt=%d ncl=%d glblSmplCnt=%d startCnt=%d[-%u,+%u] samples=[%d %d] windows=[%d %u] crtSmplIntrvlCnt=%u",
          _count, _methodInfo->getNextCompileLevel(), _totalSampleCount, _startSampleCount,
          _bodyInfo->getOldStartCountDelta(), _bodyInfo->getHotStartCountDelta(),
          _globalSamples, _globalSamplesInHotWindow,
          _scorchingSampleInterval, _hotSampleInterval, _crtSampleIntervalCount);
+      _curMsg += ((0 < msgLen) && ((size_t)msgLen <= maxMsgLen)) ? msgLen : maxMsgLen;
```

Comment:
I think another cast is warranted in this pattern, so both alternates have the same type:
```c
      _curMsg += ((0 < msgLen) && ((size_t)msgLen <= maxMsgLen)) ? (size_t)msgLen : maxMsgLen;
```